src/plugins/release-merge-purifier/index.js

```diff
@@ -0,0 +1,239 @@
+ * @private
+ */
+function getCommitMessageForPR(allCommits, context) {
+    return allCommits.data.length === 1
+        ? allCommits.data[0].commit.message
+        : context.payload.pull_request.title;
+}
+
+/**
+ * Check if the version is atleast minor
+ * @param {string} version - version to check
+ * @returns {boolean} true if its minor version
+ * @private
+ */
+function atLeastMinorVersion(version) {
+    return semver.patch(semver.clean(version)) === 0;
+}
+
+/**
+ * Get the sha value from the latest commit
+ * @param {Object} allCommits - all commit data
+ * @returns {string} latest commit sha
+ * @private
+ */
+function pluckLatestCommitSha(allCommits) {
+    return allCommits.data[allCommits.data.length - 1].sha;
+}
+
+/**
+ * Gets all the open PR
+ * @param {Object} context - context object
+ * @returns {Array<Object>} collection of release objects
+ * @private
+ */
+function getAllOpenPRs(context) {
+    return context.github.pullRequests.getAll(
+        context.issue({
+            state: "open"
+        })
+    );
+}
+
+/**
+ * Create status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} state - state can be either success or failure
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createStatusOnPR(context, state, sha) {
+    return context.github.repos.createStatus(
+        context.repo({
+            sha,
+            state,
+            target_url: "",
+            description: "semver-patch merge only phase",
+            context: "Release monitor"
+        })
+    );
+}
+
+/**
+ * Create success status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createSuccessPRStatus(context, sha) {
+    return createStatusOnPR(context, "success", sha);
+}
+
+/**
+ * Create failure status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createFailurePRStatus(context, sha) {
+    return createStatusOnPR(context, "failure", sha);
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {string} number - pr number
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function getAllCommitForPR(context, number) {
+    return context.github.pullRequests.getCommits(
+        context.repo({
+            number
+        })
+    );
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+async function createStatusOnAllPRs(context, isSuccess) {
```

Comment:
Nitpick: I think using an object argument would also be useful here, because it would avoid the [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap).

Reply:
Sounds good.